core/blockchain/blockchain_gateway.py

```diff
@@ -0,0 +1,355 @@
+import asyncio
+import base58
+import json
+import logging
+import os
+import requests
+import time
+
+import ipfsapi
+
+from core.configuration import ConfigurationManager
+from core.utils.event_types import MessageEventTypes
+
+
+logging.basicConfig(level=logging.DEBUG,
+    format='[BlockchainGateway] %(message)s')
+
+
+class BlockchainGateway(object):
+    """
+    Blockchain Gateway 
+
+    The blockchain client exposes `setter` and `getter` in order to interact
+    with the blockchain.
+
+    In order for this to work, the following must be running:
+        IPFS Daemon: `ipfs daemon`
+        The lotion app: `node app_trivial.js` from the application root directory
+
+    """
+
+    def __init__(self, config_manager, communication_manager):
+        """
+        TODO: Refactor dependencies
+        TODO: deal with config
+        """
+        # TODO: `communication_manager` is only used in a subset of methods,
+        # consider separating
+        self.communication_manager = communication_manager
+        # TODO: We will do this once the Communication Manager is done.
+        # self.communication_manager.configure_listener(self)
+
+        config = config_manager.get_config() if config_manager else {}
+        self.state = []
+        self.host = config.get("BLOCKCHAIN", "host")
+        self.ipfs_port = config.get("BLOCKCHAIN", "ipfs_port")
+        self.port = config.get("BLOCKCHAIN", "http_port")
+        self.client = None
+        try:
+            self.client = ipfsapi.connect(self.host, self.ipfs_port)
+        except Exception as e:
+            logging.info("IPFS daemon not started, got: {0}".format(e))
+
+        self.CALLBACKS = {
+            MessageEventTypes.NEW_WEIGHTS.name: self.broadcast_new_weights, 
+            MessageEventTypes.NOTHING.name: self._do_nothing,
+        }
+
+    ##########################################################################
+    ###                            API SECTION                             ###
+    ##########################################################################
+
+    def _do_nothing(self, payload: dict):
+        """
+        Do nothing.
+        """
+        pass
+
+    def get_global_state(self):
+        """
+        Gets the global state which should be a list of dictionaries
+        """
+        timeout = time.time() + 5
+        tx_receipt = None
+        while time.time() < timeout:
+            try:
+                tx_receipt = requests.get(
+                    "http://localhost:{0}/state".format(self.port))
+                tx_receipt.raise_for_status()
+                if tx_receipt:
```

Comment:
To clarify, is line 80 checking that `tx_receipt` is not `None`? If so, then wouldn't line 79 error if `tx_receipt` was `None`?

Reply:
yes line 80 is redundant. but 79 would not error because `try` would catch `requests.exceptions.ConnectionError`